core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -433,6 +434,13 @@ public static GeoDistanceSortBuilder fromXContent(QueryParseContext context, Str
                     nestedFilter = context.parseInnerQueryBuilder();
                 } else {
                     // the json in the format of -> field : { lat : 30, lon : 12 }
+                    if (fieldName != null) {
```

Comment:
Nit: maybe we could make this `fieldName.equals(currentName) == false`, so this won't fail when same name is used multiple times. Might potentially happen when generating queries?